x11-misc/lightdm/lightdm-1.32.0-r3.ebuild

```diff
@@ -0,0 +1,169 @@
+# Copyright 1999-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools pam qmake-utils readme.gentoo-r1 systemd vala xdg-utils
+
+DESCRIPTION="A lightweight display manager"
+HOMEPAGE="https://github.com/canonical/lightdm"
+SRC_URI="https://github.com/canonical/lightdm/releases/download/${PV}/${P}.tar.xz
+	mirror://gentoo/introspection-20110205.m4.tar.bz2"
+
+LICENSE="GPL-3 LGPL-3"
+SLOT="0"
+KEYWORDS="~alpha amd64 arm arm64 ~loong ppc ppc64 ~riscv x86"
+IUSE="audit elogind +gnome +gtk +introspection non-root qt5 systemd vala X"
+
+REQUIRED_USE="^^ ( elogind systemd )
+	vala? ( introspection )"
+
+RESTRICT="test"
+
+COMMON_DEPEND="
+	>=dev-libs/glib-2.44.0:2
+	dev-libs/libgcrypt:=
+	dev-libs/libxml2
+	sys-libs/pam
+	x11-libs/libX11
+	>=x11-libs/libxklavier-5
+	audit? ( sys-process/audit )
+	gnome? ( sys-apps/accountsservice )
+	introspection? ( >=dev-libs/gobject-introspection-1 )
+	qt5? (
+		dev-qt/qtcore:5
+		dev-qt/qtdbus:5
+		dev-qt/qtgui:5
+	)
+"
+RDEPEND="${COMMON_DEPEND}
+	acct-group/lightdm
+	acct-group/video
+	acct-user/lightdm
+	>=sys-auth/pambase-20101024-r2
+	elogind? ( sys-auth/elogind[pam] )
+	systemd? ( sys-apps/systemd[pam] )"
+DEPEND="${COMMON_DEPEND}
+	gnome? ( gnome-base/gnome-common )
+	X? ( x11-base/xorg-drivers )
```

Comment:
This looks rather dubious.  Why is it needed, especially in DEPEND?  Maybe add it to `optfeature` instead?

Reply:
The main reason I've added it as a depend rather than optfeature, is that it comes up so often in support that I'm starting to think we are handling this issue incorrectly to warrant adding it as a depend in the ebuild, as the users IMO, seem to want it to work this way.

Reply:
Then add it to RDEPEND, at least players will have to have it installed.  DEPEND is only required at build time and does not get installed from binpkg.

Reply:
An no USE flag please, at least for now.  I don't think lightdm can be used without xorg anyway (wayland support is still work in progress as far as I know)

Reply:
[Rationale](https://projects.gentoo.org/qa/policy-guide/dependencies.html#pg0001)

Reply:
RDEPEND makes much more sense in hindsight, thanks!